torch_geometric/profile/profile.py

```diff
@@ -195,3 +198,26 @@ def rename_profile_file(*args):
         timeline_file += '-' + arg
     timeline_file += '.json'
     os.rename('timeline.json', timeline_file)
+
+
+@contextmanager
+def e2e_time():
+    if torch.cuda.is_available():
+        torch.cuda.synchronize()
```

Comment:
I don't think this will do anything if you're not using CUDA events?

Reply:
Here is to record the time for both CPU and GPU. In GPU, it needs `torch.cuda.synchronize()` to guarantee the accuracy of time.